xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -1271,7 +1271,12 @@ void DX::DeviceResources::SetHdrColorSpace(const DXGI_COLOR_SPACE_TYPE colorSpac
   {
     if (SUCCEEDED(swapChain3->SetColorSpace1(colorSpace)))
     {
-      m_IsTransferPQ = (colorSpace == DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020);
+      bool newIsTransferPQ = (colorSpace == DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020);
+
+      if (!m_IsTransferPQ && newIsTransferPQ)
```

Comment:
Don't think its necessary local variable as only switches to PQ "one time" at start of playback. I mean use directly:

```c++
if (m_IsTransferPQ)
  DX::Windowing()->CacheSystemSdrPeakLuminance();
```

Reply:
yes good enough, it would be unlikely to switch between different "flavors" of HDR, and even then I don't know if the flow would make it here